lib/os/heap.c

```diff
@@ -290,9 +329,40 @@ void *sys_heap_aligned_alloc(struct sys_heap *heap, size_t align, size_t bytes)
 	}
 
 	set_chunk_used(h, c, true);
+
+	if (c0 == h->top_available) {
+		update_top_available(h);
+	}
+
+	sys_heap_print_stats(heap);
+
 	return mem;
 }
 
+void heap_print_stats(struct z_heap *h)
+{
+	printk("Heap stats for %p | Free: %u (min %u) "
+	       "| Contiguous: %u (min %u) | Total: %u\n", h,
+	       h->avail_chunks * CHUNK_UNIT,
+	       h->avail_chunks_min * CHUNK_UNIT,
+	       h->avail_contig * CHUNK_UNIT,
+	       h->avail_contig_min * CHUNK_UNIT,
+	       h->len * CHUNK_UNIT);
+}
+
+void sys_heap_print_stats(struct sys_heap *heap)
+{
+	heap_print_stats(heap->heap);
+}
```

Comment:
Not sure why this wraps the other function?  Doesn't seem like heap_print_stats() has another caller.